crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -0,0 +1,865 @@
+use super::{events::meta_to_event_index, Metrics, Postgres};
+use anyhow::{anyhow, bail, Context, Result};
+use contracts::cowswap_onchain_orders::{
+    event_data::OrderPlacement as ContractOrderPlacement, Event as ContractEvent,
+};
+use database::{
+    byte_array::ByteArray, events::EventIndex, onchain_broadcasted_orders::OnchainOrderPlacement,
+    orders::Order, PgTransaction,
+};
+use ethcontract::{Event as EthContractEvent, H160};
+use futures::{stream, StreamExt};
+use itertools::multiunzip;
+use model::{
+    order::{BuyTokenDestination, OrderKind, OrderUid, SellTokenSource},
+    signature::SigningScheme,
+    DomainSeparator,
+    {app_id::AppId, order::OrderData},
+};
+use number_conversions::u256_to_big_decimal;
+use shared::{
+    db_order_conversions::{
+        buy_token_destination_into, order_kind_into, sell_token_source_into, signing_scheme_into,
+    },
+    event_handling::EventStoring,
+    order_quoting::{OrderQuoting, Quote, QuoteSearchParameters},
+    order_validation::{
+        convert_signing_scheme_into_quote_kind, get_quote_and_check_fee,
+        is_order_outside_market_price,
+    },
+};
+use std::collections::HashMap;
+
+pub struct OnchainOrderParser<T: Send + Sync, W: Send + Sync> {
```

Comment:
Last naming nit:
```suggestion
pub struct OnchainOrderParser<EventData, EventRow>
where
    EventData: Send + Sync,
    EventRow: Send + Sync,
{
```

This just makes it a bit more descriptive what the `T` and `W` types are for.